src/test/java/de/tum/in/www1/artemis/programmingexercise/ProgrammingExerciseTestService.java

```diff
@@ -1835,4 +1840,41 @@ void exportSolutionRepository_shouldReturnFileOrForbidden() throws Exception {
         exportSolutionRepository(exerciseRepo, HttpStatus.FORBIDDEN);
 
     }
+
+    // TEST
+    public void buildLogStatistics_unauthorized() throws Exception {
+        exercise = programmingExerciseRepository.save(ModelFactory.generateProgrammingExercise(ZonedDateTime.now().minusDays(1), ZonedDateTime.now().plusDays(7), course));
+        request.get("/api/programming-exercises/" + exercise.getId() + "/build-log-statistics", HttpStatus.FORBIDDEN, BuildLogStatisticsDTO.class);
+    }
+
+    // TEST
+    public void buildLogStatistics_noStatistics() throws Exception {
+        exercise = programmingExerciseRepository.save(ModelFactory.generateProgrammingExercise(ZonedDateTime.now().minusDays(1), ZonedDateTime.now().plusDays(7), course));
+        var statistics = request.get("/api/programming-exercises/" + exercise.getId() + "/build-log-statistics", HttpStatus.OK, BuildLogStatisticsDTO.class);
+        assertThat(statistics.getBuildCount()).isEqualTo(0);
+        assertThat(statistics.getAgentSetupDuration()).isEqualTo(null);
+        assertThat(statistics.getTestDuration()).isEqualTo(null);
+        assertThat(statistics.getScaDuration()).isEqualTo(null);
+        assertThat(statistics.getTotalJobDuration()).isEqualTo(null);
+        assertThat(statistics.getDependenciesDownloadedCount()).isEqualTo(null);
+    }
+
+    // TEST
+    public void buildLogStatistics() throws Exception {
+        exercise = programmingExerciseRepository.save(ModelFactory.generateProgrammingExercise(ZonedDateTime.now().minusDays(1), ZonedDateTime.now().plusDays(7), course));
+        var participation = createStudentParticipationWithSubmission(INDIVIDUAL);
+        var submission1 = database.createProgrammingSubmission(participation, false);
+        var submission2 = database.createProgrammingSubmission(participation, false);
+
+        buildLogStatisticsEntryRepository.save(new BuildLogStatisticsEntry(submission1, 10, 20, 30, 60, 5));
+        buildLogStatisticsEntryRepository.save(new BuildLogStatisticsEntry(submission1, 8, 15, null, 30, 0));
```

Comment:
```suggestion
        buildLogStatisticsEntryRepository.save(new BuildLogStatisticsEntry(submission2, 8, 15, null, 30, 0));
```
Unused variable should probably be used here.

Reply:
Indeed! (Although it does not really make a difference for the result, but your are of course right!)